framework/src/messaging/messaging_send.c

```diff
@@ -0,0 +1,186 @@
+/****************************************************************************
+ *
+ * Copyright 2019 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+#include <tinyara/config.h>
+#include <debug.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <mqueue.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <sys/types.h>
+#include <messaging/messaging.h>
+#include "messaging_internal.h"
+
+#define MSG_REPLY_PRIO 10
+/****************************************************************************
+ * private functions
+ ****************************************************************************/
+#ifdef CONFIG_MESSAGING_ENABLE_HEADER
+static int messaging_append_header(const char *msg, int msglen, messaging_header_t *data)
+{
+	data->sender_pid = getpid();
```

Comment:
Can receiver receive message with header properly?
I think mq_send copies wrong memory regions.
This data structure has no consecutive regions because data->msg is allocated only.